### ISSUE TITLE
space-in-brackets - no more old behaviour?

### ISSUE THREAD

**Issue by puzrin** [eslint/eslint#1250]
In 0.8.1 rule `space-in-brackets` got more tuning options. But i could not find combination, that allow pass this code:

``` javascript
'use strict';

module.exports = function a(foo) {
  var bar = 'rty';
  // this worked in 0.8.0
  foo[bar] = [ 1 ];
};
```

space-in-brackets: [ 2, always ]
